httpclient5/src/main/java/org/apache/hc/client5/http/entity/mime/FormBodyPartBuilder.java

```diff
@@ -96,12 +99,61 @@ public FormBodyPartBuilder setField(final String name, final String value) {
         return this;
     }
 
+    /**
+     * Sets the multipart mode for this builder, determining how filenames are encoded in the
+     * {@code Content-Disposition} header. The mode affects whether the {@code filename*} parameter
+     * is included for non-ISO-8859-1 filenames.
+     * <p>
+     * In {@link HttpMultipartMode#LEGACY} mode, only the {@code filename} parameter is included
+     * with the raw value, mimicking pre-RFC 7578 behavior. In {@link HttpMultipartMode#STRICT}
+     * or {@link HttpMultipartMode#EXTENDED} modes, the {@code filename*} parameter is added
+     * with RFC 5987 encoding for filenames containing non-ISO-8859-1 characters.
+     * </p>
+     * <p>
+     * If {@code mode} is {@code null}, it defaults to {@link HttpMultipartMode#STRICT}.
+     * </p>
+     *
+     * @param mode the {@link HttpMultipartMode} to use, or {@code null} for default behavior
+     * @return this builder instance for method chaining
+     * @since 5.5
+     */
+    public FormBodyPartBuilder setMode(final HttpMultipartMode mode) {
+        this.mode = mode != null ? mode : HttpMultipartMode.STRICT;
+        return this;
+    }
+
     public FormBodyPartBuilder removeFields(final String name) {
         Args.notNull(name, "Field name");
         this.header.removeFields(name);
         return this;
     }
 
+    /**
+     * Determines whether the given string can be encoded in ISO-8859-1 without loss of data.
+     * This is used to decide whether the {@code filename} parameter can be used as-is or if
+     * the {@code filename*} parameter is needed for non-ISO-8859-1 characters.
+     *
+     * @param input the string to check, must not be {@code null}
+     * @return {@code true} if the string can be encoded in ISO-8859-1, {@code false} otherwise
+     * @since 5.5
+     */
+    private static boolean canEncodeToISO8859_1(final String input) {
+        return StandardCharsets.ISO_8859_1.newEncoder().canEncode(input);
```

Comment:
@arturobernalg Let's make the method non-static re-use the encoder rather than creating a new instance with every invocation